cluster/store/store.go

```diff
@@ -416,7 +416,65 @@ func (f *Store) FindSimilarClass(name string) string {
 	return f.db.Schema.ClassEqual(name)
 }
 
-func (st *Store) Stats() map[string]string { return st.raft.Stats() }
+// Stats returns internal statistics from this store, for informational/debugging purposes only.
+// The statistics directly from raft have their keys prefixed with "raft_".
+// See https://pkg.go.dev/github.com/hashicorp/raft#Raft.Stats for the default raft stats.
+//
+// The values of "leader_address" and "leader_id" are the respective address/ID for the current
+// leader of the cluster. They may be empty strings if there is no current leader or the leader is
+// unknown.
+//
+// The value of "ready" indicates whether this store is ready, see Store.Ready.
+//
+// The value of "is_voter" indicates whether this store is a voter, see Store.IsVoter.
+//
+// The value of "open" indicates whether this store is open, see Store.open.
+//
+// The value of "bootstrapped" indicates whether this store has completed bootstrapping,
+// see Store.bootstrapped.
+//
+// The value of "candidates" is a map[string]string of the current candidates IDs/addresses,
+// see Store.candidates.
+//
+// The value of "initial_last_applied_index" is the index of the last applied command found when
+// the store was opened, see Store.initialLastAppliedIndex.
+//
+// The value of "last_applied_index" is the index of the latest update to the store,
+// see Store.lastAppliedIndex.
+//
+// The value of "db_loaded" indicates whether the DB has finished loading, see Store.dbLoaded.
+//
+// Since this is for information/debugging we want to avoid enforcing unnecessary restrictions on
+// what can go in these stats, thus we're returning map[string]any. However, any values added to
+// this map should be able to be JSON encoded.
+func (st *Store) Stats() map[string]any {
+	stats := make(map[string]any)
+
+	// Add custom stats for this store
+	currentLeaderAddress, currentLeaderID := st.LeaderWithID()
+	stats["leader_address"] = currentLeaderAddress
+	stats["leader_id"] = currentLeaderID
+	stats["ready"] = st.Ready()
+	stats["is_voter"] = st.IsVoter()
+	stats["open"] = st.open.Load()
+	stats["bootstrapped"] = st.bootstrapped.Load()
+	stats["candidates"] = st.candidates
+	stats["initial_last_applied_index"] = st.initialLastAppliedIndex
+	stats["last_applied_index"] = st.lastAppliedIndex.Load()
+	stats["db_loaded"] = st.dbLoaded.Load()
+
+	if st.raft == nil {
+		// In this case, the "raft_" prefixed keys won't exist
+		return stats
+	}
+	// Get and add the raft stats
+	raftStats := st.raft.Stats()
+	// Add the "raft_" prefix to the stats keys which come directly from raft
+	for k, v := range raftStats {
```

Comment:
For readability reasons, we can have raft statistics, as a sub-object
`stats["db_loaded"] = st.raft.Stats()`